lib/game.rb

```diff
@@ -6,14 +6,25 @@ def initialize(board:, symbols:, user_interface:)
   end
 
   def run
-    user_interface.display_board(board.rows)
-    return if board.full?
-
-    board.put(symbols.next, user_interface.get_input)
-    run
+    user_interface.display(board.rows)
+    if win?([board.rows, board.columns, board.diagonals])
```

Comment:
I wonder if `[board.rows, board.columns, board.diagonals]` could be extracted into a public method in board so game doesn't need to know these details?

Reply:
That is my next refactor! :)

Reply:
Woot!

Reply:
Hopefully with some inverted control...

Reply:
I am thinking about something along the lines of placing this in board.

def list(fn)
  fn.call(cells.map(:&symbol))
end

And then having game be able to retrieve the board representation in the format(s) it wants. I am also considering using blocks. I am attempting to invert control and have dependent classes (game) of board be able to issue a command so that they do not have to adhere to a specific format returned by the board class.

What are your thoughts on this approach?

Reply:
Hmm. I was just thinking have whatever board class return what it knows as its list (like rows, cols, diag)...

Reply:
I think that is what I will do for now, but do you think in the future that inverting control (not necessarily in this manner) would be a good design decision in a future refactor? I have been mildly obsessed with trying to think of a way to manage the dependency of game on board.

Reply:
Honestly, I don't have a strong opinion on it. I tend to talk the line of simple/clear and try not to get too fancy with things as it decreases readability/understanding

Reply:
Fair enough. Thank you!